core/services/fluxmonitor/flux_monitor.go

```diff
@@ -423,11 +452,40 @@ func (p *PollingDeviationChecker) Start() {
 		"initr", p.initr.ID,
 	)
 
+	p.setIsHibernatingStatus()
 	go p.consume()
 }
 
+func (p *PollingDeviationChecker) setIsHibernatingStatus() {
+	if p.flagsContract != nil {
+		isHibernating, err := p.isFlagRaised()
+		if err != nil {
+			panic(err)
+		}
+		p.isHibernating = isHibernating
+	}
+}
+
+func (p *PollingDeviationChecker) isFlagRaised() (bool, error) {
+	if p.flagsContract == nil {
+		return false, nil
+	}
+	callOpts := bind.CallOpts{
+		Pending: false,
+		Context: nil,
+	}
+	flags, err := p.flagsContract.GetFlags(&callOpts, []common.Address{utils.ZeroAddress, p.initr.Address})
+	if err != nil {
+		return false, err
+	}
+	return flags[0] || flags[1], nil
+}
+
 // Stop stops this instance from polling, cleaning up resources.
 func (p *PollingDeviationChecker) Stop() {
+	if p.pollTicker != nil {
+		p.pollTicker.Stop()
```

Comment:
This should probably happen in the main run loop, in the select case where the `chStop` signal is received